corpus/convert.py

```diff
@@ -109,29 +112,41 @@ class CorpusToStmJob(Job):
     Convert a Bliss corpus into a .stm file
     """
 
+    __sis_hash_exclude__ = {"non_speech_tokens": None, "punctuation_tokens": None}
+
     def __init__(
         self,
-        bliss_corpus,
-        exclude_non_speech=True,
-        remove_punctuation=True,
-        fix_whitespace=True,
-        name="",
-        tag_mapping=(),
+        bliss_corpus: Path,
+        exclude_non_speech: bool = True,
+        non_speech_tokens: Optional[List[str]] = None,
+        remove_punctuation: bool = True,
+        punctuation_tokens: Optional[Union[str, List[str]]] = None,
+        fix_whitespace: bool = True,
+        name: str = "",
+        tag_mapping: Union[Tuple[()], Tuple[str, Dict[str, str]]] = (),
```

Comment:
The typing of tag_mapping is incorrect. The way it is currently implemented it should be
```suggestion
        tag_mapping: List[Tuple[Tuple[str, str, str], Dict[int, tk.Path]]] = (),
```
Where the 3-string tuple contains ("short name", "long name", "description") of each tag. and the `Dict[int, tk.Path]` is e.g. the output of a FilterSegments*Jobs https://github.com/rwth-i6/i6_core/blob/f267011280fe60c8ae23929eb6db77019ac59a1e/corpus/filter.py#L40-L43

Arguably this is not my finest implementation and we can revise this (now or in the future).

Reply:
Originally this was the typing information: `tuple[str, dict[str, str]]` so this was wrong?

Reply:
ok thanks